pkg/apis/cluster/v1alpha1/cluster_consts.go

```diff
@@ -37,10 +37,16 @@ const (
 
 	// ReadyCondition defines the Ready condition type that summarizes the operational state of a Cluster.
 	ReadyCondition capiv1.ConditionType = "Ready"
+	// ProvisioningReason (Severity=Info) documents that the cluster is being provisioned.
+	ProvisioningReason = "Provisioning"
+	// DeletingReason (Severity=Info) documents that the cluster is being provisioned.
+	DeletingReason = "Deleting"
 	// ProviderInitializeFailedReason (Severity=Error) documents that the cluster infrastructure provider initilize failed, probabily a wrong credential.
 	ProviderInitializeFailedReason = "ProviderInitializeFailed"
 	// ClusterResourceSetProvisionFailedReason (Severity=Error) documents that the additinal Cluster API resources (ClusterResourceSet etc.) provisioning failed.
 	ClusterResourceSetProvisionFailedReason = "ClusterResourceSetProvisionFailed"
+	// PrecheckFailedReason (Severity=Error) documents that the cluster precheck is failed.
+	PrecheckFailedReason = "PrecheckFailed"
```

Comment:
can we validate some format error in webhook?

Reply:
yes, we can validate miss/wrong key in webhook.

Reply:
but we do not get a `Secret` with client in webhook.